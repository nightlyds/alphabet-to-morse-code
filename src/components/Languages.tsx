/* eslint-disable react/require-default-props */
/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable jsx-a11y/no-static-element-interactions */
import React from "react";

export type LanguagesProps = {
    changeFirstLanguage?: (value: string) => void;
    changeSecondLanguage?: (value: string) => void;
    firstLanguage?: string;
    secondLanguage?: string;
};

const Languages = ({
    changeFirstLanguage = () => {},
    changeSecondLanguage = () => {},
    firstLanguage = "ENG",
    secondLanguage = "MORSE",
}: LanguagesProps) => {
    function setFirstLanguage(
        event: React.ChangeEvent<HTMLSelectElement>
    ): void {
        const {
            target: { value },
        } = event;
        changeFirstLanguage(value);
    }
    function setSecondLanguage(
        event: React.ChangeEvent<HTMLSelectElement>
    ): void {
        const {
            target: { value },
        } = event;
        changeSecondLanguage(value);
    }
    return (
        <div className="languages">
            <select
                name="languages-list-first"
                className="languages-list-first"
                onChange={e => setFirstLanguage(e)}
            >
                <option value="ENG" disabled={secondLanguage !== "MORSE"}>
                    ENG
                </option>
                <option value="UK" disabled={secondLanguage !== "MORSE"}>
                    UK
                </option>
                <option value="RU" disabled={secondLanguage !== "MORSE"}>
                    RU
                </option>
                <option value="MORSE">MORSE</option>
            </select>
            <select
                name="languages-list-second"
                className="languages-list-second"
                onChange={e => setSecondLanguage(e)}
            >
                <option value="MORSE">MORSE</option>
                <option value="ENG" disabled={firstLanguage !== "MORSE"}>
                    ENG
                </option>
                <option value="UK" disabled={firstLanguage !== "MORSE"}>
                    UK
                </option>
                <option value="RU" disabled={firstLanguage !== "MORSE"}>
                    RU
                </option>
            </select>
        </div>
    );
};

export default Languages;
