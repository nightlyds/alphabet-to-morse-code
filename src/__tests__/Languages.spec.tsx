import React from "react";
import Languages from "../components/Languages";
import { shallow } from "enzyme";

let setUp = () => shallow(<Languages />);

describe("Languages tests", () => {
    describe("Snapshot tests", () => {
        let component: any;

        beforeEach(() => {
            component = setUp();
        });

        it("Basic snapshot test", () => {
            expect(component).toMatchSnapshot();
        });
    });
});
