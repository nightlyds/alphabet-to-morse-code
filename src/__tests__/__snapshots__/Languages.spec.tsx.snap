// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Languages tests Snapshot tests Basic snapshot test 1`] = `
<div
  className="languages"
>
  <select
    className="languages-list-first"
    name="languages-list-first"
    onChange={[Function]}
  >
    <option
      disabled={false}
      value="ENG"
    >
      ENG
    </option>
    <option
      disabled={false}
      value="UK"
    >
      UK
    </option>
    <option
      disabled={false}
      value="RU"
    >
      RU
    </option>
    <option
      value="MORSE"
    >
      MORSE
    </option>
  </select>
  <select
    className="languages-list-second"
    name="languages-list-second"
    onChange={[Function]}
  >
    <option
      value="MORSE"
    >
      MORSE
    </option>
    <option
      disabled={true}
      value="ENG"
    >
      ENG
    </option>
    <option
      disabled={true}
      value="UK"
    >
      UK
    </option>
    <option
      disabled={true}
      value="RU"
    >
      RU
    </option>
  </select>
</div>
`;
